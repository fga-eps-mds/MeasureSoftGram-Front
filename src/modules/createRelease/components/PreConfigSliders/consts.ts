const PRE_CONFIG_SLIDERS = {
  TEST_ID: {
    CONFIG_SLIDER: 'CONFIG_SLIDER_TEST_ID',
    CONFIG_INPUT: 'CONFIG_INPUT_TEST_ID'
  }
};

export default PRE_CONFIG_SLIDERS;
