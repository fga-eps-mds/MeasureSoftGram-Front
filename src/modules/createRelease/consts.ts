export const CREATE_RELEASE_STEP = {
  ReleaseInfoStep: 0,
  ReleaseSelectorStep: 1,
  CharacteristicStep: 2,
  ThresholdStep: 3,
  ReleaseGoalStep: 4
};
