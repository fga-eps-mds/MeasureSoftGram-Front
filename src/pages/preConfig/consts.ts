const CONFIG_PAGE = {
  TITLE: 'Definir Configuração de Qualidade',
  DEFAULT_MESSAGE: 'Configuração definida com sucesso',
  ERROR_MESSAGE: 'Algo deu errado tente novamente',
  SUB_TITLE:
    'Por meio da pré-configuração que é definido quais características, subcaracterísticas e medidas serão consideradas para mensurar a qualidade do software analisado. Na pré-configuração também é definido os pesos de cada uma das entidades definidas.'
};

export default CONFIG_PAGE;
