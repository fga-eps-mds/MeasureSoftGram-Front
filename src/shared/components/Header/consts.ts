export const HEADER = {
  VALUES: {
    BUTTON_OPTIONS: [
      {
        name: 'Organizações',
        path: '/'
      },
      {
        name: 'Projetos',
        path: '/projects'
      }
    ],
    IMAGE_SOURCE: '/images/svg/logo.svg'
  }
};
