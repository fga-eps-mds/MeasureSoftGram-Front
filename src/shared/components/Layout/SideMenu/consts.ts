import ContentPasteIcon from '@mui/icons-material/ContentPaste';
// import AccountTreeIcon from '@mui/icons-material/AccountTree';

export const HEADER = {
  VALUES: {
    MENU_OPTIONS: [
      {
        name: 'Produtos',
        path: '/',
        permission: ['', 'products'],
        icon: ContentPasteIcon
      }
      // {
      //   name: 'Reposit√≥rios',
      //   path: '/repositories',
      //   permission: ['repositories'],
      //   icon: AccountTreeIcon
      // }
    ],
    IMAGE_SOURCE: '/images/svg/logo.svg'
  }
};
