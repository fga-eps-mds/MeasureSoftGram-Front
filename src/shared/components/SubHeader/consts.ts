export const SUB_HEADER = {
  VALUES: {
    OVERVIEW: 'Overview',
    MESURES: 'Medidas',
    METRICS: 'MÃ©tricas',
    PATH_NAME_INDEX: 3
  }
};
