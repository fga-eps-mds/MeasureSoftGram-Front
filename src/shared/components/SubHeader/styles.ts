import Link from 'next/link';

import styled from 'styled-components';

export const Wrapper = styled.nav`
  min-height: 50px;

  background-color: #231b22;
`;

interface Button {
  isClicked: boolean;
}

export const Button = styled.span<Button>`
  height: 50px;

  cursor: pointer;

  display: flex;
  align-items: center;

  color: white;
  border-bottom: ${({ isClicked }) => (isClicked ? '3px solid white' : null)};

  :not(:first-child) {
    margin-left: 25px;
  }
`;
