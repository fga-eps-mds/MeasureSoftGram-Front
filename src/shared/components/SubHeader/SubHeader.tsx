import React from 'react';

import Link from 'next/link';

import { Box, Container, Typography } from '@mui/material';

import { useRouter } from 'next/router';
import * as Styles from './styles';
import { SUB_HEADER } from './consts';

const { OVERVIEW, MESURES, METRICS, PATH_NAME_INDEX } = SUB_HEADER.VALUES;

interface SubHeaderOptions {
  name: string;
  path: string;
  selected: boolean;
}

const routeGetter = (path: string) => {
  return path.split('/')[PATH_NAME_INDEX];
};

function SubHeader() {
  const {
    query: { projectId },
    asPath
  } = useRouter();

  const selectedPath = routeGetter(asPath);

  const options: SubHeaderOptions[] = [
    {
      name: OVERVIEW,
      path: `/projects/${projectId}/`,
      selected: selectedPath === undefined
    },
    {
      name: MESURES,
      path: `/projects/${projectId}/measures`,
      selected: selectedPath === 'measures'
    },
    {
      name: METRICS,
      path: `/projects/${projectId}/metrics`,
      selected: selectedPath === 'metrics'
    }
  ];

  return (
    <Styles.Wrapper>
      <Container>
        <Box display="flex">
          {options.map((option: SubHeaderOptions) => (
            <Link key={option.name} href={option.path}>
              <Styles.Button key={option.name} isClicked={option.selected}>
                <Typography variant="subtitle2">{option.name}</Typography>
              </Styles.Button>
            </Link>
          ))}
        </Box>
      </Container>
    </Styles.Wrapper>
  );
}

export default SubHeader;
