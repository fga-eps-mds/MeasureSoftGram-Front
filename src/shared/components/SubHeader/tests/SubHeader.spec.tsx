import '@testing-library/jest-dom';

import React from 'react';
import { render, screen } from '@testing-library/react';

import SubHeader from '../SubHeader';
import { SUB_HEADER } from '../consts';

const { OVERVIEW, MESURES, METRICS } = SUB_HEADER.VALUES;

const BUTTON_TITLE_TYPE = 'h6';

describe('<SubHeader />', () => {
  describe('Snapshot', () => {
    it('Deve corresponder ao Snapshot', () => {
      const tree = render(<SubHeader />);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Comportamento', () => {
    beforeEach(() => {
      render(<SubHeader />);
    });

    describe.each([{ name: OVERVIEW }, { name: MESURES }, { name: METRICS }])('Botões', ({ name }) => {
      it(`Deve renderizar botão ${name}`, () => {
        const button = screen.getByText(name);
        expect(button).toBeDefined();
        expect(button.nodeName.toLowerCase()).toBe(BUTTON_TITLE_TYPE);
      });
    });
  });
});
