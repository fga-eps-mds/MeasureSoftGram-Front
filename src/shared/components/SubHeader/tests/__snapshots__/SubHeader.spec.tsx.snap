// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubHeader /> Snapshot Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="styles__Wrapper-sc-7jbcpa-0 hYqiAJ"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiBox-root css-k008qs"
          >
            <span
              class="styles__Button-sc-7jbcpa-1 ipcWGP"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
              >
                Overview
              </h6>
            </span>
            <span
              class="styles__Button-sc-7jbcpa-1 iYDoZS"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
              >
                Medidas
              </h6>
            </span>
            <span
              class="styles__Button-sc-7jbcpa-1 iYDoZS"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
              >
                Métricas
              </h6>
            </span>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="styles__Wrapper-sc-7jbcpa-0 hYqiAJ"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-k008qs"
        >
          <span
            class="styles__Button-sc-7jbcpa-1 ipcWGP"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
            >
              Overview
            </h6>
          </span>
          <span
            class="styles__Button-sc-7jbcpa-1 iYDoZS"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
            >
              Medidas
            </h6>
          </span>
          <span
            class="styles__Button-sc-7jbcpa-1 iYDoZS"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
            >
              Métricas
            </h6>
          </span>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
